### DISCUSSION TITLE
Option to minify inline scripts

### DISCUSSION THREAD

**Discussion by gacek1123** [withastro/roadmap#1063] [Proposal]
### Body

# Summary

An option similar to `<script is:inline` to make Astro minify the script while still inlining it without any additional postprocessing.

# Background & Motivation

Astro currently has a set of really good features for handling `<script>` tags, but there are still times when a user needs full control over the script's structure and position. For this, you can simply add `is:inline` to the script tag, which will skip any kind of postprocessing and put the script in the final page as-is.

Sometimes, a user needs to have a long inline script, which can increase the size of the document, worsening page load times. It would be great to have an option to tell Astro to do a simple minification pass on the script using esbuild to remove unnecessary white spaces, shorten variable names, etc. without any additional changes like script deduplication or transpiling.

# Goals

- Allow users to easily and automatically minify their inline scripts during build
- Make this feature be off by default and allow users to opt-in on a per-script basis
- Possibly add new configuration options in `astro.config.js` to allow users to specify exactly what aspects they want to minify (whitespace, identifiers, syntax)
# Example

```html
<script is:inline-minified>
    // Some comment
    const text = "Hello world!";

    console.log(text);
</script>
```
Changed into:
```html
<script>const t="Hello world!";console.log(t);</script>
```

**Comment by delucis** [comment]
Thanks for the suggestion. 

Sharing this other discussion which overlaps quite a bit: https://github.com/withastro/roadmap/discussions/995
